### PR TITLE
Laboratory work #4, Polina Kobyakova - 22FPL2

### PR DESCRIPTION
**Pull request by kobyakovapolina**

### CODE REVIEW

File: lab_4_fill_words_by_ngrams/main.py [hunk 1]
```diff
@@ -98,6 +135,38 @@ def run(self, seq_len: int, prompt: str) -> str:  # type: ignore
                 or if sequence has inappropriate length,
                 or if methods used return None.
         """
+        if not (isinstance(seq_len, int) and seq_len > 0 and isinstance(prompt, str) and prompt):
+            raise ValueError
```
[the remark this answers is not in the document]
Comment (kobyakovapolina) [reply lab_4_fill_words_by_ngrams/main.py]: done

File: lab_4_fill_words_by_ngrams/main.py [hunk 2]
```diff
@@ -98,6 +137,38 @@ def run(self, seq_len: int, prompt: str) -> str:  # type: ignore
...
                 or if methods used return None.
         """
+        if not (isinstance(seq_len, int) and seq_len > 0 and isinstance(prompt, str) and prompt):
+            raise ValueError('Inappropriate input or input argument is empty or seq_len is a negative int')
+
+        encoded_text = self._word_processor.encode(prompt)
+        if not encoded_text:
+            raise ValueError('Could not encode')
+
+        for i in range(seq_len):
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:147]: Where do you use `i`?

File: lab_4_fill_words_by_ngrams/main.py [hunk 3]
```diff
@@ -114,6 +185,9 @@ def __init__(self) -> None:
         """
         Initialize an instance of GeneratorTypes.
         """
+        self.greedy = 0
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:188]: And for what did you create these fields?

File: lab_4_fill_words_by_ngrams/main.py [hunk 4]
```diff
@@ -125,6 +199,12 @@ def get_conversion_generator_type(self, generator_type: int) -> str:  # type: ig
         Returns:
             (str): Name of the generator.
         """
+        if generator_type == 0:
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:202]: use you properties!

File: lab_4_fill_words_by_ngrams/main.py [hunk 5]
```diff
@@ -56,6 +78,20 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> str:  #
...
+            raise ValueError('Inappropriate input or input argument is empty')
+
+        decoded_sentences = ''
+        sentences = ' '.join(list(decoded_corpus)).split(self._end_of_word_token)
+
+        for sentence in sentences:
+            sentence = sentence.strip().capitalize()
+            if not sentence:
+                break
+            sentence = f"{sentence}. "
```
[the remark this answers is not in the document]
Comment (kobyakovapolina) [reply lab_4_fill_words_by_ngrams/main.py:91]: fixed

File: lab_4_fill_words_by_ngrams/main.py [hunk 6]
```diff
@@ -4,6 +4,10 @@
 Top-p sampling generation and filling gaps with ngrams
 """
 # pylint:disable=too-few-public-methods, too-many-arguments
+import random
+import math
+from typing import Tuple
```
[the remark this answers is not in the document]
Comment (kobyakovapolina) [reply lab_4_fill_words_by_ngrams/main.py:9]: fixed

File: lab_4_fill_words_by_ngrams/main.py (hunk 4 above)
Comment (kobyakovapolina) [reply lab_4_fill_words_by_ngrams/main.py:202]: fixed?

File: lab_4_fill_words_by_ngrams/main.py (hunk 3 above)
Comment (kobyakovapolina) [reply lab_4_fill_words_by_ngrams/main.py:188]: fixed? I wanted it to use it for get_conversion_generator_type function.

File: lab_4_fill_words_by_ngrams/main.py (hunk 2 above)
Comment (kobyakovapolina) [reply lab_4_fill_words_by_ngrams/main.py:147]: fixed

File: lab_4_fill_words_by_ngrams/main.py [hunk 7]
```diff
@@ -125,6 +205,10 @@ def get_conversion_generator_type(self, generator_type: int) -> str:  # type: ig
         Returns:
             (str): Name of the generator.
         """
+        generator_types = {self.greedy: 'Greedy Generator', self.top_p: 'Top-P Generator',
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:208]: move to init

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_fill_words_by_ngrams/main.py (hunk 7 above)
Comment (kobyakovapolina) [reply lab_4_fill_words_by_ngrams/main.py:208]: done

**Comment by vitazaytseva** [comment]
Well done! Good job!

**Pull request closed without merging**